contracts/RMRK/extension/tokenProperties/IRMRKTokenPropertiesRepository.sol

```diff
@@ -0,0 +1,288 @@
+     * @param collection The collection address
+     * @param tokenId The token ID
+     * @param key The key of the property
+     * @param value The new value of the property
+     */
+    event StringPropertyUpdated(
+        address indexed collection,
+        uint256 indexed tokenId,
+        string key,
+        string value
+    );
+
+    /**
+     * @notice Used to notify listeners that an uint property has been updated.
+     * @param collection The collection address
+     * @param tokenId The token ID
+     * @param key The key of the property
+     * @param value The new value of the property
+     */
+    event UintPropertyUpdated(
+        address indexed collection,
+        uint256 indexed tokenId,
+        string key,
+        uint256 value
+    );
+
+    /**
+     * @notice Used to notify listeners that a boolean property has been updated.
+     * @param collection The collection address
+     * @param tokenId The token ID
+     * @param key The key of the property
+     * @param value The new value of the property
+     */
+    event BoolPropertyUpdated(
+        address indexed collection,
+        uint256 indexed tokenId,
+        string key,
+        bool value
+    );
+
+    /**
+     * @notice Used to notify listeners that an address property has been updated.
+     * @param collection The collection address
+     * @param tokenId The token ID
+     * @param key The key of the property
+     * @param value The new value of the property
+     */
+    event AddressPropertyUpdated(
+        address indexed collection,
+        uint256 indexed tokenId,
+        string key,
+        address value
+    );
+
+    /**
+     * @notice Used to notify listeners that a bytes property has been updated.
+     * @param collection The collection address
+     * @param tokenId The token ID
+     * @param key The key of the property
+     * @param value The new value of the property
+     */
+    event BytesPropertyUpdated(
+        address indexed collection,
+        uint256 indexed tokenId,
+        string key,
+        bytes value
+    );
+
+    /**
+     * @notice Used to register a collection to use the RMRK token properties repository.
+     * @dev  If the collection does not implement the Ownable interface, the `useOwnable` value must be set to `false`.
+     * @dev Emits an {AccessControlRegistration} event.
+     * @param collection The address of the collection that will use the RMRK token properties repository.
+     * @param issuer The address of the issuer of the collection.
+     * @param useOwnable The boolean value to indicate if the collection implements the Ownable interface and whether it
+     *  should be used to validate that the caller is the issuer (`true`) or to use the manually set issuer address
+     *  (`false`).
+     */
+    function registerAccessControl(
+        address collection,
+        address issuer,
+        bool useOwnable
+    ) external;
```

Comment:
Removing support for non ownable implementer also removes the need for this I think. The issuer can just start managing access control for properties directly.